src/orchestrator/src/lib.rs

```diff
@@ -258,8 +262,13 @@ pub struct CpuLimit {
 }
 
 impl CpuLimit {
+    pub const MAX: Self = Self::from_millicpus(usize::MAX / 1_000_000);
     /// Constructs a new CPU limit from a number of millicpus.
-    pub fn from_millicpus(millicpus: usize) -> CpuLimit {
+    pub const fn from_millicpus(millicpus: usize) -> CpuLimit {
+        assert!(
+            millicpus.checked_mul(1_000_000).is_some(),
+            "Nano-CPUs must be representable"
+        );
```

Comment:
What about adding a `as_nanocpu` function to avoid the overflow check in `pack_all_replica_size_updates`? This seems to encapsulate the guarantee in one place instead of two.

Reply:
Good point, I'll do that